### PR TITLE
Add more tests

### PR DESCRIPTION
**Pull request by constantinpape**
@FynnBe I tried the `export_package` functionality:
```python
spec_path = export_package(unet2d_nuclei_broad_model_url, Path("./data"))
```
but this fails with
```
            if f.name not in changes:
                if f._field_type is _FIELD_INITVAR:
>                   raise ValueError(f"InitVar {f.name!r} "
                                     'must be specified with replace()')
E                   ValueError: InitVar 'uri_string' must be specified with replace()

../../../../../software/conda/miniconda3/envs/torch-em-test/lib/python3.8/dataclasses.py:1284: ValueError

```

### CODE REVIEW

**Comment by FynnBe** [comment]
oh, beautiful! I'll check it out...

**Comment by FynnBe** [comment]
> but this fails with

quick update: part 1 of the fix is in https://github.com/bioimage-io/spec-bioimage-io/pull/200


still failing with, but now with

```
..\..\..\..\spec-bioimage-io\bioimageio\spec\shared\io.py:234: in serialize_raw_node_to_dict
    serialized = schema_class().dump(raw_node)
C:\conda\envs\selfseg\lib\site-packages\marshmallow\schema.py:552: in dump
    result = self._serialize(processed_obj, many=many)
C:\conda\envs\selfseg\lib\site-packages\marshmallow\schema.py:520: in _serialize
    value = field_obj.serialize(attr_name, obj, accessor=self.get_attribute)
C:\conda\envs\selfseg\lib\site-packages\marshmallow\fields.py:318: in serialize
    return self._serialize(value, attr, obj, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <fields.RelativeLocalPath(default=<marshmallow.missing>, attribute=None, validate=[<Attribute(attribute='suffix', is_g... 'validator_failed': 'Invalid value.', 'invalid': 'Not a valid string.', 'invalid_utf8': 'Not a valid utf-8 string.'})>
value = URI(scheme='https', authority='raw.githubusercontent.com', path='bioimage-io/spec-bioimage-io/main/example_specs/models/unet2d_nuclei_broad/README.md', query='', fragment='')
attr = 'documentation'
obj = Model(attachments={}, authors=[Author(name='Constantin Pape;@bioimage-io', affiliation='EMBL Heidelberg', orcid='0000-...path='bioimage-io/spec-bioimage-io/main/example_specs/models/unet2d_nuclei_broad/weights.pt', query='', fragment=''))})
kwargs = {}

    def _serialize(self, value, attr, obj, **kwargs) -> typing.Optional[str]:
        if value is not None:
>           assert isinstance(value, pathlib.Path), value
E           AssertionError: https://raw.githubusercontent.combioimage-io/spec-bioimage-io/main/example_specs/models/unet2d_nuclei_broad/README.md

..\..\..\..\spec-bioimage-io\bioimageio\spec\shared\fields.py:342: AssertionError
```

this is probablh due to the incomplete packaging and also will need fixing in spec-bioimage-io....

**Comment by FynnBe** [comment]
fixed in https://github.com/bioimage-io/spec-bioimage-io/pull/199
(will be merged to main in: https://github.com/bioimage-io/spec-bioimage-io/pull/196)

File: tests/weight_converter/torch/test_torchscript.py [hunk 1]
```diff
@@ -1,8 +1,7 @@
-from pathlib import Path
 from bioimageio.spec import export_package
 
 
 # TODO only run this test if we have pytorch
 def test_torchscript_converter(unet2d_nuclei_broad_model_url):
-    spec_path = export_package(unet2d_nuclei_broad_model_url, Path("./data"))
```
Comment (FynnBe) [inline tests/weight_converter/torch/test_torchscript.py]: it doesn't make sense to specify a root path here, the root_path is meant only to resolve relative paths in a given rdf (if the file is given as raw node or 'content dict'/'yaml.load(rdf_file)')

the returned path is in the BioImage.IO cache folder (in tmp)... do we need to add an output path argument?

Comment (constantinpape) [reply tests/weight_converter/torch/test_torchscript.py]: > it doesn't make sense to specify a root path here, the root_path is meant only to resolve relative paths in a given rdf (if the file is given as raw node or 'content dict'/'yaml.load(rdf_file)')

Ok, please add that to the documentation; and I think you could even raise a `ValueError` for invalid combination, i.e. url + root path, with a meaningful error message.

> the returned path is in the BioImage.IO cache folder (in tmp)

that's good here, as we just want to use the data for the tests.

> do we need to add an output path argument?

yes, I think that makes sense, but it should be optional (and if not given return the cache path)

Comment (FynnBe) [reply tests/weight_converter/torch/test_torchscript.py]: > Ok, please add that to the documentation; and I think you could even raise a ValueError for invalid combination, i.e. url + root path, with a meaningful error message.

new issue: https://github.com/bioimage-io/spec-bioimage-io/issues/202

> yes, I think that makes sense, but it should be optional (and if not given return the cache path)

new issue: https://github.com/bioimage-io/spec-bioimage-io/issues/203

**Comment by constantinpape** [comment]
Now the error is 
```
    def __post_init__(self):
        from . import nodes, raw_nodes
    
        if not isinstance(self, (nodes.Node, raw_nodes.RawNode)):
>           raise TypeError("base nodes should not be instantiated!")
E           TypeError: base nodes should not be instantiated!

../../../../spec-bioimage-io/bioimageio/spec/shared/base_nodes.py:33: TypeError
```

**Comment by constantinpape** [comment]
Now it's working.

**Comment by constantinpape** [comment]
There is still an issue in the Frunet tests:
```
    def http_error_default(self, req, fp, code, msg, hdrs):                                                                                                                                                         
>       raise HTTPError(req.full_url, code, msg, hdrs, fp)                                                                                                                                                          
E       urllib.error.HTTPError: HTTP Error 404: Not Found                                                                                                                                                           
                                                                                                                                                                                                                    
../../../software/conda/miniconda3/envs/torch-em-test/lib/python3.8/urllib/request.py:649: HTTPError                                                                                                                
------------------------------------------------------------------------------------------------ Captured log setup ------------------------------------------------------------------------------------------------
WARNING  bioimageio.spec.shared.fields:fields.py:357 invalid local relative path: https://raw.githubusercontent.com/deepimagej/models/master/fru-net_sev_segmentation/auto_converted_documentation.md               
ERROR    download:utils.py:355 Failed to download https://raw.githubusercontent.com/deepimagej/models/master/fru-net_sev_segmentation/auto_converted_documentation.md
```
So it looks like the model yaml is not quite correct.

**Comment by constantinpape** [comment]
@FynnBe the problem here is that this model has a html as documentation: https://github.com/deepimagej/models/blob/master/fru-net_sev_segmentation/model.yaml#L14 
I guess this is auto-converted somehow, but then the auto-converted url doesn't exist?!

**Comment by FynnBe** [comment]
> I guess this is auto-converted somehow, but then the auto-converted url doesn't exist?!

Yes, the autoconvert atm is only capable of passing the static test... I'll fix it...

**Comment by FynnBe** [comment]
https://github.com/bioimage-io/spec-bioimage-io/pull/210 should fix it...

**Comment by constantinpape** [comment]
The error still persists, see full stack trace below
@FynnBe it's very easy to reproduce it, just check out this branch and run 
```
pytest -k test_build_spec_keras
```

Stack trace
```
../conftest.py:16:                                                                                                                                                                                                  
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../spec-bioimage-io/bioimageio/spec/delegator.py:147: in export_resource_package                                                                                                                              
    return io_cls.export_resource_package(                                                                                                                                                                          
../../../spec-bioimage-io/bioimageio/spec/shared/io_.py:322: in export_resource_package                                                                                                                             
    package_content = cls.get_resource_package_content(                                                                                                                                                             
../../../spec-bioimage-io/bioimageio/spec/model/v0_3/utils.py:100: in get_resource_package_content                                                                                                                  
    raw_rd = incl_as_local(raw_rd, "documentation")                                                                                                                                                                 
../../../spec-bioimage-io/bioimageio/spec/model/v0_3/utils.py:92: in incl_as_local                                                                                                                                  
    fp = resolve_uri(value, root_path=root_path)                                                                                                                                                                    
../../../../software/conda/miniconda3/envs/torch-em-test/lib/python3.8/functools.py:875: in wrapper                                                                                                                 
    return dispatch(args[0].__class__)(*args, **kw)                                                                                                                                                                 
../../../spec-bioimage-io/bioimageio/spec/shared/utils.py:236: in _resolve_uri_uri_node                                                                                                                             
    local_path = _download_uri_to_local_path(path_or_remote_uri)                                                                                                                                                    
../../../spec-bioimage-io/bioimageio/spec/shared/utils.py:353: in _download_uri_to_local_path                                                                                                                       
    urlretrieve(str(uri), str(local_path))                                                                                                                                                                          
../../../../software/conda/miniconda3/envs/torch-em-test/lib/python3.8/urllib/request.py:247: in urlretrieve                                                                                                        
    with contextlib.closing(urlopen(url, data)) as fp:                                                                                                                                                              
../../../../software/conda/miniconda3/envs/torch-em-test/lib/python3.8/urllib/request.py:222: in urlopen                                                                                                            
    return opener.open(url, data, timeout)                                                                                                                                                                          
../../../../software/conda/miniconda3/envs/torch-em-test/lib/python3.8/urllib/request.py:531: in open                                                                                                               
    response = meth(req, response)                                                                                                                                                                                  
../../../../software/conda/miniconda3/envs/torch-em-test/lib/python3.8/urllib/request.py:640: in http_response                                                                                                      
    response = self.parent.error(                                                                                                                                                                                   
../../../../software/conda/miniconda3/envs/torch-em-test/lib/python3.8/urllib/request.py:569: in error
    return self._call_chain(*args)
../../../../software/conda/miniconda3/envs/torch-em-test/lib/python3.8/urllib/request.py:502: in _call_chain
    result = func(*args)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <urllib.request.HTTPDefaultErrorHandler object at 0x7f6d116d5d60>, req = <urllib.request.Request object at 0x7f6d116cc280>, fp = <http.client.HTTPResponse object at 0x7f6d116ddc10>, code = 404
msg = 'Not Found', hdrs = <http.client.HTTPMessage object at 0x7f6d116ddaf0>

    def http_error_default(self, req, fp, code, msg, hdrs):
>       raise HTTPError(req.full_url, code, msg, hdrs, fp)
E       urllib.error.HTTPError: HTTP Error 404: Not Found

../../../../software/conda/miniconda3/envs/torch-em-test/lib/python3.8/urllib/request.py:649: HTTPError
------------------------------------------------------------------------------------------------ Captured log setup ------------------------------------------------------------------------------------------------
WARNING  bioimageio.spec.shared.fields:fields.py:357 invalid local relative path: https://raw.githubusercontent.com/deepimagej/models/master/fru-net_sev_segmentation/auto_converted_documentation.md
ERROR    download:utils.py:355 Failed to download https://raw.githubusercontent.com/deepimagej/models/master/fru-net_sev_segmentation/auto_converted_documentation.md
```

**Comment by FynnBe** [comment]
requires https://github.com/bioimage-io/spec-bioimage-io/pull/211

**Comment by constantinpape** [comment]
This is still failing (with https://github.com/bioimage-io/spec-bioimage-io/pull/211), but it looks like it is a different error now:
```
tests/build_spec/test_build_spec.py:51: in _test_build_spec                                                                                                                                                         
    raw_model = build_model(**kwargs)                                                                                                                                                                               
bioimageio/core/build_spec/build_model.py:418: in build_model                                                                                                                                                       
    model = model_spec.schema.Model().load(serialized)
E           marshmallow.exceptions.ValidationError: {'documentation': ['expected relative path.']}q
```

**Comment by FynnBe** [comment]
> This is still failing (with bioimage-io/spec-bioimage-io#211), but it looks like it is a different error now:

I suspect this might be fixed with packagin the model before validating it as the auto-converted documentation is not a local relative path... 
I'll try this out tomorrow.

**Comment by constantinpape** [comment]
@FynnBe any progress here? would be good to merge this so that we can use the models also in other tests.

**Comment by constantinpape** [comment]
Looks like there is a problem with the dependency now:
```
node = 'conda:environment.yaml', field_name = 'file'                                                                                                                                                                                                   
                                                                                                                                                                                                                                                       
    def incl_as_local(node: _GenericRawNode, field_name: str) -> _GenericRawNode:                                                                                                                                                                      
>       value = getattr(node, field_name)                                                                                                                                                                                                              
E       AttributeError: 'str' object has no attribute 'file'                                                                                                                                                                                           
                                                                                                                                                                                                                                                       
../spec-bioimage-io/bioimageio/spec/model/v0_3/utils.py:85: AttributeError              
```

**Comment by constantinpape** [comment]
The tests vs. spec.main are passing, I am merging this.

**Pull request merged**